Source/DafnyLanguageServer/Workspace/TextDocumentLoader.cs

```diff
@@ -84,34 +79,13 @@ IOptions<DafnyPluginsOptions> compilerOptions
     }
 
     public async Task<DafnyDocument> LoadAsync(TextDocumentItem textDocument, CancellationToken cancellationToken) {
-      var request = new LoadRequest(textDocument, cancellationToken);
-      requestQueue.Add(request, cancellationToken);
-      return await request.Document.Task;
+#pragma warning disable CS1998
+      return await await Task.Factory.StartNew(async () => LoadInternal(textDocument, cancellationToken), cancellationToken,
```

Comment:
Can you get rid of an `await` ? I am not sure but I don't see a reason why to use an `async` closure here.
```suggestion
      return await Task.Factory.StartNew(() => LoadInternal(textDocument, cancellationToken), cancellationToken,
```